man/man8/zfs-load-key.8

```diff
@@ -166,6 +166,27 @@ will become one. Alternatively, the
 .Fl i
 flag may be provided to cause an encryption root to inherit the parent's key
 instead.
+.Pp
+Changing the user's key does not change the underlying master key used to
+encrypt the dataset.  Therefore, if a user key is compromised,
```

Comment:
How about .. "Therefore, if a user key is compromised, and a change-key is executed, newly-written data will continue to be encrypted with the same master key.  Due to Copy-on-Write or uncertainty of hardware operations (ie. sector remapping and wear leveling algorithms), the previous user key will remain on disk until overwritten.  At any time, possession of the master key itself, or a former user key along with its complementary wrapped master key, obtainable through forensic methods, are avenues for an attack."

Reply:
That's a great explanation to add, but I would like to keep the high-level impact up front.  I.e. keep the first 2 sentences of this section as-is: `Changing the user's key does not change the underlying master key used to
 encrypt the dataset.  Therefore, if a user key is compromised,
zfs change-key does not necessarily protect newly-written data from attack.`

Or we could reorder the sentences so that this paragraph reads `If a user key is compromised,
zfs change-key does not necessarily protect newly-written data from attack.  This is because newly-written data will continue to be encrypted with the same master key. Due to ...`

Reply:
@ahrens :  I like the reorder, but think we could re-enforce the vulnerability of "existing" data

Three things I was trying to achieve:
1) "change key" protects neither existing or newly-written data. The 2nd sentence only speaks to newly-written.

2) clearly delineate "user key", "wrapped master key", and "master key"

3) offer the reasoning that ZFS cannot guarantee prior user key disposal

Reply:
I reworded this as:

If the user's key is compromised, zfs change-key does not necessarily protect existing or newly-written data from attack.  Newly-written data will continue to be encrypted with the same master key as the existing data.  The master key is compromised if an attacker obtains a user key and the corresponding wrapped master key. Currently, zfs change-key does not overwrite the previous wrapped master key on disk, so it is accessible via forensic analysis for an indeterminate length of time.

Reply:
Thanks.  This seems to hit all the goals.  I like it.